### PR TITLE
New function to generate Person Accounts

### PR DESCRIPTION
**Pull request by prescod**
* Generates Person Accounts in the format expected by CCI

Also:

* New concept of a Parse-time Macro Plugin
* New testing fixture function
* Shorter syntax for Plugin imports
* Update link to CCI docs
* Add a sortedcontainers package for ordered sets

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -1525,8 +1518,133 @@ recipe once. It could, however, save you time if you  were running
...
+
+You could make a file named `do_gooders/plugins/salesforce/org/DoGood.py`
+
+And that file would contain a class like this:
+
+```python
+from snowfakery.plugins import SnowfakeryPlugin
+
+class DoGood(SnowfakeryPlugin):
+    """Plugin which generates a summataion helper"""
```
Comment (davisagli) [inline docs/index.md:1642]:
```suggestion
    """Plugin which generates a summation helper"""
```

**Review by davisagli — COMMENTED** [review]
I'm not really convinced this needs such a special-purpose plugin and new type of macro. I feel like it obscures the simplicity of the data structure that will be generated. Why not just show an example in the docs of creating person accounts using the existing functionality and point out which fields are crucial for it to work?

**Comment by prescod** [comment]
@davisagli I was also tempted to "just document it" but the documentation would not meet my standards of usability/readability.

```yaml
    PersonContactId:
      - object: Contact
        fields:
          IsPersonAccount: "true"
          AccountId:
            reference: Account
```

CumulusCI insists on the value being the literal string "true", not True, "True" or 1. Making CumulusCI more forgiving would need to happen in a couple of places, including SQL queries. I anticipated pushback and complicated CumulusCI for a fix that is only really needed by Snowfakery (and other, potential side-loading tools, of which there are currently none).

This would be the only place in Snowfakery where truth is represented by the string "true".

Another issue is that it would be very tempting to add additional fields to the Contact object above but those fields would never make it into Salesforce.

**Comment by davisagli** [comment]
@prescod I'd call that a bug in CumulusCI that should be fixed. It supports multiple representations of true in other places when loading.

**Comment by davisagli** [comment]
> Another issue is that it would be very tempting to add additional fields to the Contact object above but those fields would never make it into Salesforce.

This should feel normal to people who have any experience loading person accounts via the API. They just need a reminder in the docs to put the contact fields on the Account.

File: snowfakery/generate_mapping_from_recipe.py [hunk 2]
```diff
@@ -26,7 +43,7 @@ def build_dependencies(intertable_dependencies):
     dependencies = {}
     reference_fields = {}
     for dep in intertable_dependencies:
-        table_deps = dependencies.setdefault(dep.table_name_from, set())
+        table_deps = dependencies.setdefault(dep.table_name_from, SortedSet())
```
[the remark this answers is not in the document]
Comment (prescod) [reply snowfakery/generate_mapping_from_recipe.py:46]: The intent is for this to replace most sets in Snowfakery, or at least newly coded sets.

As I've documented elsewhere, I don't think that the price of using sets should be that you have to risk the determinacy of your program and risk "it works on my computer but not yours" type problems. I have seen those problem where mapping files are generated in a different order at different times. The difference "should" be benign, but if it isn't then I have a hard-to-track-down bug.

The dependency is pretty small and I think its worth it.

[the remark this answers is not in the document]
Comment (prescod) [reply snowfakery/generate_mapping_from_recipe.py]: https://github.com/SFDO-Tooling/Snowfakery/pull/283/commits/e7f08b75037f4a5c418d22fb8267f534a97c0537

**Comment by prescod** [comment]
Nobody had a strong feeling about it but most had a slight preference for the Plugin syntax. I could at some point make the Plugin declaration optional which would probably also make at least one ambivalent respondent more enthusiastic.

**Review by davisagli — APPROVED** [review]

**Pull request merged**